clang/test/SemaHLSL/resource_binding_attr_error_resource.hlsl

```diff
@@ -0,0 +1,76 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.3-library -x hlsl -o - -fsyntax-only %s -verify
+
+// This test validates the diagnostics that are emitted when a variable with a "resource" type
+// is bound to a register using the register annotation
+
+// expected-error@+1  {{binding type 'b' only applies to constant buffer resources}}
+RWBuffer<int> a : register(b2, space1);

```

Comment:
This is testing RWBuffer instead of the underlying resource class attributes.

Reply:
I think this still needs to be addressed? It looks like we still have some tests using RWBuffer directly.

Reply:
I don't think we should remove all instances of RWBuffer in testing. Although we plan to eventually move all hlsl resources into an appropriate header, with the corresponding spellable hlsl attributes, as it stands now RWBuffer is defined in HLSLExternalSemaSource.cpp within a template specialization decl. There is code in this change that specifically accounts for this, and by removing RWBuffer entirely from the tests, it would become dead code. 
Specifically, `getSpecifiedHLSLAttrFromRecordDecl` has an if condition to check if the record decl is a `ClassTemplateSpecializationDecl`, which is only the case for types declared in HLSLExternalSemaSource.cpp. 
This change should test both user defined types that have the spellable attribute of interest, but ideally the end goal is that it works with actual practical HLSL resources, like RWBuffer.

Reply:
I think that the ClassTemplateSpecializationDecl is going to become dead anyway once we put the attribute in the right place.  This is why I think we should set these up so that the attribute is in the correct place now rather than later because we're doing work now to exercise code that we know is going to almost immediately change.

Reply:
Again, if you really don't want to do that now, we need an issue to clean this up properly.  It's a shame to be introducing technical debt like this.

Reply:
Ok, I'll remove RWBuffer, replace them with a UDT that has the SRV attribute, and remove the dead code.